streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java

```diff
@@ -412,16 +576,19 @@ public void consume(String topic) {
         while (true) {
             ConsumerRecords<Integer, byte[]> records = consumer.poll(500);
             if (records.isEmpty()) {
-                if (endKey.equals(key))
+                if (key.compareTo(endKey) >= 0)
```

Comment:
Is it possible that `key.compareo(endKey) > 0`? Just trying to not hide any potential issues with relaxed condition.